### ISSUE TITLE
Cascade Delete Issue

### ISSUE THREAD

**Issue by jbertman** [collerek/ormar#90]
I can't seem to get cascade deletes working with FKs on a postgres DB. Should it work by default?

I tried adding the ondelete="CASCADE" kwarg to my FK relations, but nothing seemed to happen on the other side of the model. 

In short, I have:
```python
class Project(BaseModel):
    class Meta(ormar.ModelMeta):
        tablename = "projects"

    name: str = ormar.String(max_length=256)
    # <SNIP>

class Quote(BaseModel):
    class Meta(ormar.ModelMeta):
        tablename = "quotes"

    # <SNIP>
    project: Project = ormar.ForeignKey(
        Project, related_name="quotes", ondelete="CASCADE"
    )
```

And the routing endpoint I have in FastAPI:
```python
@router.delete("/{project_id}")
async def delete_project(request: Request, project_id: uuid.UUID):
    project = await Project.objects.filter(client=request.state.client).get(
        id=project_id
    )
    await project.delete()
    return {"deleted": True, "id": project.id}
```
But this seems to orphan the affected quote that was attached to the project. This looks to be a known caveat to SQLAlchemy when using filter().delete(). Is there an easy way to build in the same fixes?
https://stackoverflow.com/questions/5033547/sqlalchemy-cascade-delete
https://stackoverflow.com/questions/19243964/sqlalchemy-delete-doesnt-cascade

**Comment by collerek** [comment]
That's not the case as `ormar` uses only core part of the `sqlalchemy`, so the one to create tables and build selects/inserts etc. So most of the issues for `sa` won't help you with `ormar` as they usually refer to `orm` part (the one that you inherit from base etc.). You need to specifically search for core issues but even this might not help you as `ormar` is built on top of this.

When you call `get` or `all` in `ormar` you actually get the results back and later you `delete` this particular instance. If you want to delete without loading there is `delete` also in `QuerySet`, i recently updated the docs in this regard.

Will check what's going on with the cascades.

**Comment by jbertman** [comment]
@collerek I tried to generate a new revision after adding `ondelete`, but no changes were detected by alembic. I can try to do it from scratch and report back.

**Comment by collerek** [comment]
There is something wrong i investigate what exactly. In sqlite it sometimes does not create the flat. Can you check if the database structure in your case includes foreign keys definition? Or just the columns and no relations whatsoever?

**Comment by jbertman** [comment]
AFAICT, it does not appear to create any FK relations or constraints on the
DB for Postgres

**Comment by jbertman** [comment]
While not ideal, I solved this manually by adding the constraints to my Alembic script as such (still using the above example):

```python
    op.create_table(
        "projects",
        sa.Column("id", sa.CHAR(36), nullable=False),
        sa.Column("created_date", sa.DateTime(), nullable=True),
        sa.Column("updated_date", sa.DateTime(), nullable=True),
        sa.Column("name", sa.String(length=256), nullable=False),
        # <SNIP>
    )
    op.create_table(
        "quotes",
        sa.Column("id", sa.CHAR(36), nullable=False),
        sa.Column("created_date", sa.DateTime(), nullable=True),
        sa.Column("updated_date", sa.DateTime(), nullable=True),
        # <SNIP>
        sa.PrimaryKeyConstraint("id"),
        sa.ForeignKeyConstraint(
            ["project"], ["projects.id"], ondelete="CASCADE", onupdate="CASCADE"
        ),
    )
```
That allows everything else to function as normal, and lets the DB take care of cascading the operation (can't speak to support for other backends, but this works for postgres).

**Comment by collerek** [comment]
Yeah, but that how it's supposed to look like out of the box :)

I found the bug it's related to deep-cloning sqlalchemy.Columns, will fix this and release new version soon.

Also will add names to foreign_keys so alembic can clearly identify constraints and not recreate them during consecutive runs.

**Comment by collerek** [comment]
Fixed in 0.9.0 by #92
